builtin/logical/aws/path_static_roles.go

```diff
@@ -202,14 +205,38 @@ func (b *backend) pathStaticRolesDelete(ctx context.Context, req *logical.Reques
 }
 
 func (b *backend) validateRoleName(name string) error {
+	if name == "" {
+		return errors.New("empty role name attribute given")
+	}
 	return nil
 }
 
 func (b *backend) validateIAMUserExists(ctx context.Context, req *logical.Request, username string) error {
+	c, err := b.clientIAM(ctx, req.Storage)
+	if err != nil {
+		return fmt.Errorf("unable to validate username '%s': %w", username, err)
+	}
+
+	// we don't really care about the content of the result, just that it's not an error
+	out, err := c.GetUser(&iam.GetUserInput{
+		UserName: aws.String(username),
+	})
+	if err != nil || out.User == nil || *out.User.UserName != username {
+		return fmt.Errorf("unable to validate username '%s': %w", username, err)
```

Comment:
nit: wondering if we should split the last case of the usernames not matching into a separate error with a different, more specific error string to provide more info in case of failures?

Reply:
We can, sure.

I don't expect that clause to trigger unless something is going very wrong (or perhaps someone is trying something tricky) so perhaps that makes it worth its own callout

Reply:
I've separated this out to another check, and added a test case